sdk/src/main/java/com/bugsnag/android/FileStore.java

```diff
@@ -82,26 +100,63 @@ String write(@NonNull T streamable) {
                 filename), exception);
         } finally {
             IOUtils.closeQuietly(out);
+            lock.unlock();
         }
         return null;
     }
 
-    @NonNull abstract String getFilename(T streamable);
+    @NonNull
+    abstract String getFilename(T streamable);
 
     List<File> findStoredFiles() {
```

Comment:
I think this can be called multiple times in the current design, which might cause bad things to happen. Potentially this should only return files that are not already in the queued files set?

Reply:
Good point. I've updated this method so that it only returns stored files which aren't in the `queuedFiles` collection.